src/Domain/Bootstrap.php

```diff
@@ -100,11 +100,10 @@ protected function init() {
 		$this->register_dependencies();
 		$this->register_payment_methods();
 
-		if ( is_admin() ) {
-			if ( $this->package->get_version() !== $this->package->get_version_stored_on_db() ) {
-				$this->migration->run_migrations();
-				$this->package->set_version_stored_on_db();
-			}
+		// This is just a temporary solution to make sure the migrations are run. We have to refactor this. More details: https://github.com/woocommerce/woocommerce-blocks/issues/10196.
+		if ( $this->package->get_version() !== $this->package->get_version_stored_on_db() ) {
```

Comment:
Just noting here that on every page `get_version_stored_on_db()` will run `get_option()`. There was conversations around performance but given that this is a temporary fix, and that `get_option()` has caching this was a trade off we were willing to temporarily take.